homeassistant/components/light/mqtt/schema_basic.py

```diff
@@ -114,11 +114,73 @@
     config.setdefault(
         CONF_STATE_VALUE_TEMPLATE, config.get(CONF_VALUE_TEMPLATE))
 
-    async_add_entities([MqttLight(
-        config.get(CONF_NAME),
-        config.get(CONF_UNIQUE_ID),
-        config.get(CONF_EFFECT_LIST),
-        {
+    async_add_entities([MqttLight(config, discovery_hash)])
+
+
+class MqttLight(MqttAvailability, MqttDiscoveryUpdate, Light, RestoreEntity):
+    """Representation of a MQTT light."""
+
+    def __init__(self, config, discovery_hash):
+        """Initialize MQTT light."""
+        self._state = False
+        self._sub_state = None
+        self._brightness = None
+        self._hs = None
+        self._color_temp = None
+        self._effect = None
+        self._white_value = None
+        self._supported_features = 0
+
+        self._name = None
```

Comment:
This should be updated to just store `config` on the instance

Reply:
@balloob Sure, I'll open a follow-up which does that once this is merged.
Let me know if you prefer to include those changes here as well.